packages/trip-form/src/MetroModeSelector/index.tsx

```diff
@@ -0,0 +1,257 @@
+    strategy,
+    middlewareData: { arrow: { x: arrowX, y: arrowY } = {} }
+  } = useFloating({
+    open,
+    onOpenChange: setOpen,
+    middleware: [offset(8), shift(), arrow({ element: arrowRef })]
+  });
+
+  const modeButtonClicked = () => {
+    if (disableHover) {
+      if (!modeButton.enabled) {
+        // enable mode button if it's off
+        onToggle();
+      }
+    } else {
+      onToggle();
+    }
+  };
+
+  const { getFloatingProps, getReferenceProps } = useInteractions([
+    useHover(context, {
+      enabled: !disableHover,
+      handleClose: safePolygon({
+        blockPointerEvents: false,
+        restMs: 500,
+        buffer: 0
+      })
+    }),
+    useClick(context, { enabled: disableHover }),
+    useRole(context),
+    useDismiss(context),
+    { reference: { onClick: modeButtonClicked } }
+  ]);
+
+  const disableModeButton = () => {
+    if (modeButton.enabled) {
+      onToggle();
+    }
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <ModeButtonItem
+        className={modeButton.enabled ? "enabled" : ""}
+        ref={reference}
+        role="checkbox"
+        // This library relies on prop spreading
+        /* eslint-disable-next-line react/jsx-props-no-spreading */
+        {...getReferenceProps()}
+        aria-expanded={null}
+        aria-checked={modeButton.enabled}
+        aria-label={modeButton.label}
+        fillModeIcons={fillModeIcons || fillModeIcons !== false}
+      >
+        <modeButton.Icon size={32} />
+      </ModeButtonItem>
+      <FloatingPortal
+        root={floatingTarget}
+        id="otp-ui-metro-mode-selector-hover"
+      >
+        {open && modeButton.enabled && (
+          <HoverPanel
+            // This library relies on prop spreading
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...getFloatingProps()}
+            ref={floating}
+            style={{
+              position: strategy,
+              top: y ?? 0,
+              left: x ?? 0
+            }}
+          >
+            <Arrow
+              ref={arrowRef}
+              style={{ top: arrowY ?? 0, left: arrowX ?? 0 }}
+            />
+            <HoverInnerContainer>
+              <SubSettingsPane
+                modeButton={modeButton}
+                onSettingUpdate={onSettingsUpdate}
+                showControls={disableHover}
+                onDismiss={() => {
+                  setOpen(false);
+                }}
+                onDisableMode={disableModeButton}
+              />
+            </HoverInnerContainer>
+          </HoverPanel>
+        )}
+      </FloatingPortal>
+    </>
+  );
+}
+
+interface Props {
+  disableHover?: boolean;
+  modeButtons?: ModeButtonDefinition[];
+  onSettingsUpdate: (QueryParamChangeEvent) => void;
+  onToggleModeButton: (key) => void;
+  fillModeIcons?: boolean;
+}
+
+export default function ModeSelector({
+  onToggleModeButton,
+  onSettingsUpdate,
+  modeButtons = [],
+  disableHover,
+  fillModeIcons
```

Comment:
Add a story for storybook to demonstrate what this prop does.

Reply:
I see that Storybook provides a "knob" for this prop, but it doesn't seem that the prop has any effect.